### PR TITLE
chore(argo-cd): Move notification configs

### PR DESCRIPTION
**Pull request by pdrastil**
Resolves: https://github.com/argoproj/argo-helm/issues/1381

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Comment by pdrastil** [comment]
Little bit different implementation of https://github.com/argoproj/argo-helm/pull/1299

File: charts/argo-cd/templates/argocd-configs/argocd-notifications-cm.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ include "argo-cd.notifications.configMapName" . }}
+  name: argocd-notifications-cm
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-configs/argocd-notifications-cm.yaml:5]: Why hardcoding CM name?

Comment (pdrastil) [reply charts/argo-cd/templates/argocd-configs/argocd-notifications-cm.yaml:5]: Aligning with upstream
https://github.com/argoproj/argo-cd/blob/master/manifests/base/notification/argocd-notifications-cm.yaml

Comment (pdrastil) [reply charts/argo-cd/templates/argocd-configs/argocd-notifications-cm.yaml:5]: Plus the hardcoded names in Argo CLI

**Review by mkilchhofer — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

**Pull request merged**